tools/yul_phaser/main.cpp

```diff
@@ -15,7 +15,69 @@
 	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-int main([[maybe_unused]] int argc, [[maybe_unused]] char** argv)
+
+#include <boost/program_options.hpp>
+
+#include <iostream>
+#include <string>
+
+using namespace std;
+namespace po = boost::program_options;
+
+void runAlgorithm(string const& _sourcePath)
+{
+	cout << "Input: " << _sourcePath << endl;
+}
+
+int main(int argc, char** argv)
 {
+	po::options_description description(
+		"yul-phaser, a tool for finding the best sequence of Yul optimisation phases.\n"
+		"\n"
+		"Usage: yul-phaser [options] <file>\n"
+		"Reads <file> as Yul code and tries to find the best order in which to run optimization"
+		" phases using a genetic algorithm.\n"
+		"Example:\n"
+		"yul-phaser program.yul\n"
+		"\n"
+		"Allowed options",
+		po::options_description::m_default_line_length,
+		po::options_description::m_default_line_length - 23
+	);
+
+	description.add_options()
+		("help", "Show help message and exit.")
+		("input-file", po::value<string>()->required(), "Input file")
+	;
+
+	po::positional_options_description positionalDescription;
+	po::variables_map arguments;
+	positionalDescription.add("input-file", 1);
+	po::notify(arguments);
+
+	try
+	{
+		po::command_line_parser parser(argc, argv);
+		parser.options(description).positional(positionalDescription);
+		po::store(parser.run(), arguments);
+	}
+	catch (po::error const & _exception)
+	{
+		cerr << _exception.what() << endl;
+		return 1;
+	}
+
+	if (arguments.count("help") > 0) {
+		cout << description << endl;
+		return 2;
+	}
+
+	if (arguments.count("input-file") == 0) {
+		cerr << "Missing argument: input-file." << endl;
+		return 1;
+	}
```

Comment:
Może, żeby trzymać jednakowy poziom abstrakcji w cały main warto byłoby wyciągnąć helpa i parsowanie argumentów do osobnej funkcji?

Reply:
Can we keep reviews in English please?